File: frontend/lib/utils/shipment_couriers.dart
const defaultShipmentCouriers = ['In-house', 'DPD', 'Royal Mail'];

List<String> shipmentCourierOptionsFromSettings(String? raw) {
  final lines = (raw ?? '')
      .split('\n')
      .map((l) => l.trim())
      .where((l) => l.isNotEmpty)
      .toList();
  return lines.isNotEmpty ? lines : List<String>.from(defaultShipmentCouriers);
}

File: management-frontend/src/utils/shipmentCouriers.ts
export const DEFAULT_SHIPMENT_COURIERS = ['In-house', 'DPD', 'Royal Mail'];

/** One courier name per line from company settings; falls back to defaults when empty. */
export function shipmentCourierOptionsFromSettings(raw?: string | null): string[] {
  const lines = (raw ?? '')
    .split('\n')
    .map((l) => l.trim())
    .filter((l) => l.length > 0);
  return lines.length > 0 ? lines : [...DEFAULT_SHIPMENT_COURIERS];
}
